graspologic/layouts/auto.py

```diff
@@ -11,205 +11,539 @@
+        the lowest weights will be pruned until at most ``max_edges`` exist. Warning:
+        this pruning is approximate and more edges than are necessary may be pruned.
+        Running in 32 bit environment you will most likely need to reduce this number
+        or you will run out of memory.
+
+    Returns
+    -------
+    nx.Graph
+        The largest connected component of the graph after pruning edges if applicable.
+    """
+    graph = _approximate_prune(graph, max_edges)
+    graph = largest_connected_component(graph)
+    return graph
 
-    In addition to pruning edges by weight, this function also only operates over the
-    largest connected component in the graph.
 
-    After dimensionality reduction, sizes are generated for each node based upon
-    their degree centrality, and these sizes and positions are further refined by an
-    overlap removal phase. Lastly, a global partitioning algorithm
-    (:func:`graspologic.partition.leiden`) is executed for the largest connected
-    component and the partition ID is included with each node position.
+def embed_for_layout(
+    graph: nx.Graph,
+    algorithm: str = "n2v",
+    validated: bool = False,
+    dimensions: int = 128,
+    num_walks: int = 10,
+    window_size: int = 2,
+    iterations: int = 3,
+    n_components: Optional[int] = None,
+    svd_algorithm: str = "randomized",
+    random_seed: Optional[int] = None,
+    embed_kwds={},
+) -> Tuple[np.array, np.array]:
+    """
+    Generates a node2vec embedding, an adjacency spectral embedding, or a laplacian
+    spectral embedding from a given graph.
 
     Parameters
     ----------
     graph : :class:`networkx.Graph`
-        The graph to generate a layout for. This graph may have edges pruned if the
-        count is too high and only the largest connected component will be used to
-        automatically generate a layout.
-    perplexity : int
-        The perplexity is related to the number of nearest neighbors that is used in
+        The graph for which the embedding is generated. Only the largest connected component
+        will be used to calculate the embedding.
+    algorithm: str (default="n2v")
+        The embedding algorithm to be used.
+            - 'n2v'
+                Generates embedding using :func:`~graspologic.embed.node2vec_embed`
+            - 'ase'
+                Generates embedding using :class:`~graspologic.embed.AdjacencySpectralEmbed`
+            - 'lse'
+                Generates embedding using :class:`~graspologic.embed.LaplacianSpectralEmbed`
+    validated: bool (default=False)
+        Whether the graph has been checked if it is fully connected in the undirected case
+        or weakly connected in the directed case. If False and the input is not connected,
+        the function will use its largest connected component to generate an embedding.
+    dimensions: int (default=128)
+        Applicable to node2vec embedding. Dimensionality of the word vectors.
```

Comment:
I vote we not use two different names for `n_components`/`dimensions` in the same function, so we could just use `n_components here`